packages/examples/phases/src/utils.tsx

```diff
@@ -0,0 +1,25 @@
+import React, {FC, useEffect} from 'react';
+
+export function deferredLoadComponent<T>(
+  Resource: DeferredImport<T extends {default: any} ? T : never>,
+): FC {
+  let loaded = false;
+  let cleanUp: undefined | (() => void);
+  return function WrappedComponent(props) {
+    useEffect(() => {
+      return () => {
+        cleanUp?.();
+      };
+    }, []);
+    if (loaded) {
+      return <Resource.mod.default {...props} />;
```

Comment:
We'll need to document that we require a default export. Alternatively we could pass in a key to map to the correct import, but that's messy business I'd like to avoid